runners/google-cloud-dataflow-java/worker/src/test/java/org/apache/beam/runners/dataflow/worker/testing/GenericJsonAssert.java

```diff
@@ -28,19 +29,23 @@
   private static final JacksonFactory jacksonFactory = JacksonFactory.getDefaultInstance();
 
   /**
-   * Asserts that {@code actual} has the same JSON representation as {@code expectedJsonText}.
+   * Asserts that {@code actual} has the same JSON representation as {@code expected}.
    *
-   * @param expectedJsonText expected JSON string.
+   * @param expected expected JSON string, {@link GenericJson}, {@link java.util.Map}, or {@link
+   *     Iterable} of {@link GenericJson}.
    * @param actual actual object to compare its JSON representation.
    */
-  public static void assertEqualsAsJson(String expectedJsonText, GenericJson actual) {
-    actual.setFactory(jacksonFactory);
+  public static void assertEqualsAsJson(Object expected, Object actual) {
```

Comment:
Initially I thought about creating overridden methods for `assertEqualsAsJson(String, GenericJson)`, `assertEqualsAsJson(Iterable, Iterable)`. `assertEqualsAsJson(Map, GenericJson)`, and so on. But it turned out that these implementations are almost same. So declaring just `assertEqualsAsJson(Object, Object)`.